app/src/components/Results.js

```diff
@@ -22,16 +26,28 @@ function Results({ apiUrl }) {
       setResults(data.data.data);
     });
   };
-  const clickDetail = (id) => {
-    //跳转详情页,拼接路由动态参数
-    history.push({ pathname: `/parkDetails/${id}` });
+
+  const getUserAllfavorites = (id) => {
+    //to query user's saved parks
+    axios
+      .get(
+        `http://localhost:4000/api/tasks/getUserFavorites?userId=${user.sub}`,
+      )
+      .then((res) => {
+        setSavedIds(res.data.data.map((item) => item.park_id)); //update savedIds
+      });
   };
```

Comment:
Maybe you want to put the shared parts of`getUserAllfavorites()` function in a helper module, to share between `Results.js` and `Park.js`.